lifecycle-operator/controllers/lifecycle/keptnworkloadversion/controller_test.go

```diff
@@ -1143,16 +1143,15 @@ func setupReconciler(objs ...client.Object) (*KeptnWorkloadVersionReconciler, ch
 
 	fakeClient := fake.NewClient(objs...)
 
-	recorder := record.NewFakeRecorder(100)
-	r := &KeptnWorkloadVersionReconciler{
-		Client:        fakeClient,
-		Scheme:        scheme.Scheme,
-		EventSender:   controllercommon.NewK8sSender(recorder),
-		Log:           ctrl.Log.WithName("test-appController"),
-		TracerFactory: tf,
-		Meters:        controllercommon.InitAppMeters(),
-		SpanHandler:   &telemetry.SpanHandler{},
+	SchedulingGatesHandler := &fake.ISchedulingGatesHandlerMock{
+		EnabledFunc: func() bool {
+			return false
+		},
 	}
+
+	recorder := record.NewFakeRecorder(100)
+	r := NewReconciler(fakeClient, scheme.Scheme, controllercommon.NewK8sSender(recorder), ctrl.Log.WithName("test-appController"),
```

Comment:
the unit tests should now make use of a mocked evaluation handler, now that we have the injectable interface for the reconciler

Reply:
Alright

Reply:
I didn't not Implement evaluationHandler in the test as `NewReconciler` fills in the parameter needed by the evaluationHandler here https://github.com/keptn/lifecycle-toolkit/pull/2299/files/55964f8e94d5b6e419441ba46b1b408d32f65aa5#diff-7e3a3dd6a5940394c614f51b5abf528919d2593178aad9cfbbe963bdc261dc6bR77-R90

Reply:
please do use the mocked implementation here, as this is the main purpose of introducing the evaluationhandler interface. In the unit tests we should only validate the behavior of the controller itself and how it behaves when e.g. the evaluation handler returns an unexpected error. This is much harder when using the actual implementation of the evaluationHandler interface, as you would need to find a set of parameters for the reconciler that results in an error when calling the evaluationHandler's methods.

Reply:
Alright I will look into that.